src/mutations/appAction.ts

```diff
@@ -27,7 +27,7 @@ export default (queryConfig: QueryClientConfig) => {
           const newData: AppAction = newAppAction;
           // check that the websocket event has not already been received and therefore the data were added
           if (!prevData) {
-            queryClient.setQueryData(key, newData);
+            queryClient.setQueryData(key, [newData ?? []]);
```

Comment:
```suggestion
            // we need to wrap the created appAction in an array because the cache key will receive all the actions but the post call only return the current posted data
            queryClient.setQueryData<AppAction[]>(key, [newData]);
```